scriptHelper.js

```diff
@@ -1,43 +1,176 @@
+  missionTarget.innerHTML = `<h2>Mission Destination</h2>
+          <ol>
+            <li>Name: ${name}</li>
+            <li>Diameter: ${diameter}</li>
+            <li>Star: ${star}</li>
+            <li>Distance from Earth: ${distance} </li>
+           <li>Number of Moons: ${moons}</li>
+          </ol>
+          <img src="${imageUrl}">`;
 }
+//addDestinationInfo(pickPlanets(planets));
 
 function validateInput(testInput) {
-   
+  //validateInput() should take in a string as a parameter and return "Empty", "Not a Number", or "Is a Number" as appropriate.
+  if (testInput === "") {
+    return "Empty";
+  } else if (!isNaN(testInput)) {
+    return "Is a Number";
+  } else if (typeof String(testInput) === "string") {
+    return "Not a Number";
+  }
 }
 
+//you will use validateInput() to complete the formSubmission() function.
 function formSubmission(document, list, pilot, copilot, fuelLevel, cargoLevel) {
-   
+  //console.log("formSubmission");
+  //if any of these input variables is blank, throw and alert
+  if (
+    validateInput(pilot) === "Empty" ||
+    validateInput(copilot) === "Empty" ||
+    validateInput(fuelLevel) === "Empty" ||
+    validateInput(cargoLevel) === "Empty"
+  ) {
+    alert("All fields are required!");
+  }
+  //Pilot and CoPilot Names should be a String, and Fuel Level and Cargo Mass should be a Number
+  if (
+    validateInput(pilot) !== "Not a Number" ||
+    validateInput(copilot) !== "Not a Number" ||
+    validateInput(fuelLevel) !== "Is a Number" ||
+    validateInput(cargoLevel) !== "Is a Number"
+  ) {
+    //alert the user
+    alert("Enter text for names and numbers for fuel and cargo levels");
+  }
+  //if Fuel and Cargo are in good range, then it meets all the requirements, set the InnerHTML to say its ready to launch - show the list as visible before all the checks
+
+  if (fuelLevel >= 10000 && cargoLevel <= 10000) {
+    // make faulty items list visible.
+    list.style.visibility = "visible";
+    //If the shuttle is ready to launch, change the text of launchStatus to green and display "Shuttle is ready for launch".
+    let launchStatus = document.getElementById("launchStatus");
+    launchStatus.innerHTML = "Shuttle is ready for launch";
+    launchStatus.style.color = "green";
+
+    //tell it what to show for the list items
+    let pilotStatus = document.getElementById("pilotStatus");
+    pilotStatus.innerHTML = `Pilot, ${pilot}, is Ready for Launch`;
+
+    let copilotStatus = document.getElementById("copilotStatus");
+    copilotStatus.innerHTML = `CoPilot, ${copilot}, is Ready for Launch`;
+
+    let fuelStatus = document.getElementById("fuelStatus");
+    fuelStatus.innerHTML = "Fuel level high enough for launch";
+
+    let cargoStatus = document.getElementById("cargoStatus");
+    cargoStatus.innerHTML = "Cargo mass low enough for launch";
+    //test if both fuel and cargo are out of required bounds
+  } else if (fuelLevel <= 10000 && cargoLevel >= 10000) {
```

Comment:
```suggestion
  } else if (fuelLevel < 10000 && cargoLevel > 10000) {
```
Use the less-than sign instead of less-than-or-equal-to sign on line 84 in `scriptHelper.js`. So, this line will read in pseudocode would be this: 'if the fuel level is less than 10000 && cargo level is greater than 10000'. These if statements should check if these values are above 10000 since it is an acceptable level for fuel and cargo levels.